apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/main/java/gov/cms/bfd/pipeline/rda/grpc/RdaSink.java

```diff
@@ -83,16 +84,24 @@ default int writeMessage(String dataVersion, TMessage object) throws ProcessingE
    * @param apiVersion The version of the api used to get the message.
    * @param message The message that was being transformed when the error occurred.
    * @param exception The exception that was thrown while transforming the message.
-   * @throws IOException If there was an issue writing out the error.
+   * @throws IOException If there was an issue writing the error out
+   * @throws ProcessingException If there was a problem processing the claim
    */
   default void writeError(
       String apiVersion, TMessage message, DataTransformer.TransformationException exception)
-      throws IOException {
+      throws IOException, ProcessingException {
     throw new UnsupportedOperationException();
   }
 
   /**
-   * Write all of the objects to the data store and return the number of objects actually written.
+   * Checks if the error limit has been exceeded.
+   *
+   * @throws ProcessingException If the error limit was reached.
+   */
+  void checkErrorCount() throws ProcessingException;
```

Comment:
Since this is only called from within `writeError` it does not need to be in the `RdaSink` interface.  It's just an implementation detail.

Alternatively if it's left in the interface it should be called right after `writeError` by users of the `RdaSink`.  I can see an argument for `writeError` not throwing if it successfully writes the error since the write did not itself fail. I think it should be one or the other though.

Reply:
Never mind I see it being called from `retrieveAndProcessObjects`.

Reply:
This is actually also called from `StandardGrpcRdaSource` to check it before running a job: https://github.com/CMSgov/beneficiary-fhir-data/pull/1565/files/29ae213fd33ca7453a40a97268865c011e13334f#diff-1ca50d94e4cca6e1b14974692b5ce9bcb0324bf5955bd2b90ac4e17d6b4f5b2eR179